elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -473,8 +483,15 @@ export class PeerPool extends EventEmitter {
 		return this.getAllPeers().map(peer => peer.peerInfo);
 	}
 
-	public getAllPeers(): ReadonlyArray<Peer> {
-		return [...this._peerMap.values()];
+	public getAllPeers(
+		kind?: typeof OutboundPeer | typeof InboundPeer,
```

Comment:
This should be a string enum 'outbound' or 'inbound' for consistency. Also this is confusing because in JavaScript `typeof OutboundPeer` would evaluate to 'function'.

Reply:
What was the conclusion about this one? Although in tyepescript, this seems to be valid

Reply:
We have no better option and it is valid Typescript-wise.

Reply:
Because this method is only used internally by the PeerPool and not exposed publicly, I was convinced that it's OK in this case.